torchtune/models/clip/_position_embeddings.py

```diff
@@ -100,23 +100,314 @@ def __init__(
 
         self.gate = nn.Parameter(torch.zeros(1))
 
+        self._register_load_state_dict_pre_hook(self._load_state_dict_hook)
+
+    def _load_state_dict_hook(
```

Comment:
I don't see any test case for this, which arguably is the important part

Reply:
it mostly calls the other functions. But it makes sense to create a unit test for it.